metrics-operator/controllers/common/providers/prometheus/common.go

```diff
@@ -40,13 +41,21 @@ func (r RoundTripperRetriever) GetRoundTripper(ctx context.Context, provider met
 		}
 		return nil, err
 	}
-	return config.NewBasicAuthRoundTripper(secret.User, secret.Password, "", "", promapi.DefaultRoundTripper), nil
+
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: provider.Spec.InsecureSkipTlsVerify, 
+		},
+	}
+
+	return config.NewBasicAuthRoundTripper(secret.User, secret.Password, "", "", transport), nil
```

Comment:
Please make the transport variable of type `RoundTripper` and include also the parameters and values which are present in the `DefaultRoundTripper`, so we do not break the existing behavior.

Reply:
Hey @odubajDT I have used **promapi.DefaultRoundTripper** to maintain the current behavior and used **TLSClientConfig** for TLS handling

Reply:
I would prefer using the roudtripper as a parameter instead of only the tlsconfig, please have a look how the `DefaultRoundTripper` is build and add the tls config to it. Also please run and adapt the unit tests

Reply:
Hey @odubajDT 
I have made changes now Instead of always creating a new transport I have modified to preserve behaviour of DefaultRoundTripper

Reply:
the logic can be simplified, you can just assign the value from the CR directly to the tlsconfig, without any cloning and conditions

Reply:
let me look into this

Reply:
@odubajDT so what I understand is that we can directly use `InsecureSkipTlsVerify` value from the `provider.Spec `to the TLS config

Reply:
Hey @odubajDT I have improved the logic and aligned the test according to the current test

Reply:
please resolve the failing unit tests

Reply:
@odubajDT The failing unit tests is fixed now..